File: beginners/2-problemset-condicionais/C++/1036-formula-bhaskara/formula_bhaskara.cpp
#include <iostream>
#include <iomanip>
#include <math.h>
using namespace std;

int main() {
    double A, B, C, delta, raiz1, raiz2;

    cin >> A >> B >> C;

    delta = pow(B, 2) - (4 * A * C);
    raiz1 = ((-B) + sqrt(delta)) / (2 * A);
    raiz2 = ((-B) - sqrt(delta)) / (2 * A);

    if (A == 0.0 || B == 0.0 || C == 0.0 || delta < 0.0)
        cout << "Impossivel calcular" << endl;
    else {
        cout << "R1 = " << fixed << setprecision(5) << raiz1 << endl;
        cout << "R2 = " << fixed << setprecision(5) << raiz2 << endl;
    }

    return 0;
}

File: beginners/2-problemset-condicionais/Python/1036-formula-bhaskara/formula_bhaskara.py
from math import sqrt

a, b, c = input().split()
a = float(a)
b = float(b)
c = float(c)

try:
    delta = (b ** 2) - 4 * a * c
    r1 = (-b + sqrt(delta)) / (2 * a)
    r2 = (-b - sqrt(delta)) / (2 * a)
    print(f'R1 = {r1:.5f}\nR2 = {r2:.5f}')
except (ValueError, ZeroDivisionError):
    print('Impossivel calcular')
